test/edu/csus/ecs/pc2/core/UtilitiesTest.java

```diff
@@ -621,4 +621,60 @@ public void testCopy() throws Exception {
 
     }
 
+    
+    
+    public void testLocatingSampleFiles() throws Exception {
+
+        IInternalContest contest = loadFullSampleContest(null, "mini");
+        assertNotNull(contest);
+        
+        ContestInformation info = contest.getContestInformation();
+        
+        String judgeCDP = info.getJudgeCDPBasePath();
+        
+        Problem[] problems = contest.getProblems();
+        
+        int totFiles = 0;
+        for (Problem problem : problems) {
+            totFiles += problem.getNumberTestCases();
+        }
+        
+        assertEquals("Expecting total data and sample files",10, totFiles);
+        
+        for (Problem problem : problems) {
+
+            ProblemDataFiles problemDataFiles = contest.getProblemDataFile(problem);
+
+            SerializedFile[] judgeFiles = problemDataFiles.getJudgesDataFiles();
+            for (SerializedFile judgeDataFile : judgeFiles) {
+
+                String dataFileLocation = Utilities.locateJudgesDataFile(problem, judgeDataFile, judgeCDP, DataFileType.JUDGE_DATA_FILE);
+                assertFileExists(dataFileLocation, "Could not find datafile");
```

Comment:
It would be helpful if the output message (when the above assertion fails) included the name of the data file which could not be found.  (Presumably that is available as `judgeDataFile.getName()` (or something similar), although it might be necessary to add an `assertNotNull(judgeDataFile)` check too.)